tests/src/main/scala/BasicQueryTests.scala

```diff
@@ -299,7 +364,8 @@ final class BasicJoinQueryTests(implicit xa: Transactor[IO]) {
   val bookSchema: TableQuery[Book] =
     caseClassSchema(
       TableName("books"),
-      (column[Long]("id"), column[Long]("user_id"), column[Option[Long]]("parent_id")).mapN(Book[Reference])
+      (column[Long]("id"), column[Long]("user_id"), column[Long]("parent_id").map(Reference.liftOption(_)), column[String]("name"))
```

Comment:
Again, this will need to be based on something different than Reference (preferably a universally quantified type that we know nothing about, or just a sealed one with no public methods), so that it's impossible to e.g. map on the reference or mess up the future `all` method.